net/lp_rpc.proto

```diff
@@ -211,16 +214,21 @@ message NotifySegment {
     // URL of the segment to transcode.
     string url      = 1;
 
-    // Job the segment belongs to.
-    string job      = 2;
+    // Configuration for the transcoding job
+    SegData segData = 3;
 
     // ID for this particular transcoding task.
     int64 taskId   = 16;
 
-    // Set of profiles to transcode this segment into.
+// Deprecated fields. May still be populated sometimes for back compat
+
+    // Deprecated by segData. Job the segment belongs to.
+    string job      = 2;
+
+    // Deprecated by fullProfiles. Set of presets to transcode into.
     bytes profiles = 17;
```

Comment:
recommended way of deprecating fields
```suggestion
    bytes profiles = 17 [deprecated=true];
```

Reply:
Very nice - didn't know about this annotation! Actually, kept this line in place since we still "use" it via setting an invalid value to induce failures. Added a comment to the protobuf indicating this in 4d957856cdf5136a1e21972169e9f50d64e244c4

According to the protobuf docs, this is a no-op in all languages except Java, where it emits a warning. My concern is that if golang ever picks this up - golang only has errors, no warnings -  this could cause code to stop compiling and the annotation would have to be removed. I'm not entirely sure about the implications of this, so it seems safer to just keep things as-is.